spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,252 @@
+require_relative "spec_helper"
+
+describe "ReservationManager" do
+  let(:reservation_manager) {
+    Hotel::ReservationManager.new
+  }
+  describe "initialize" do
+    it "can be instantiated" do
+      expect(reservation_manager).must_be_kind_of Hotel::ReservationManager
+    end
+
+    it "establishes the base structures when instantiated" do
+      [:rooms, :reservations, :blocks].each do |prop|
+        expect(reservation_manager).must_respond_to prop
+      end
+
+      expect(reservation_manager.rooms).must_be_kind_of Array
+      expect(reservation_manager.reservations).must_be_kind_of Array
+      expect(reservation_manager.blocks).must_be_kind_of Array
+    end
+
+    it "has 20 rooms" do
+      expect(reservation_manager.rooms.length).must_equal 20
+    end
+  end
+
+  describe "reserve method" do
+    let(:new_reservation) {
+      reservation_manager.reserve(room_id: 1,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+    }
+    it "can reserve an available room and adds a new reservation to the list of reservations" do
+      before_reserve = reservation_manager.reservations.length
+      new_reservation
+      after_reserve = reservation_manager.reservations.length
+      expect(after_reserve - before_reserve).must_equal 1
+    end
+
+    it "cannot reserve unavailable rooms" do
+      new_reservation
+      expect { reservation_manager.reserve(new_reservation) }.must_raise ArgumentError
```

Comment:
You are feeding into the arguments of `reserve` the results of a previous `reserve` call.  That doesn't make sense.  

Instead I would suggest 2 lines, one `new_reservation` and the other a full `reservation_manager.reserve` call with the same dates and room number.